backend/LeetWars.Core/LeetWars.Core.BLL/Services/UserService.cs

```diff
@@ -43,4 +47,21 @@ public async Task<bool> CheckIsExistingEmail(string email)
         bool isExistingEmail = await _context.Users.AnyAsync(u => u.Email.ToLower() == email.ToLower());
         return isExistingEmail;
     }
+
+    public async Task<UserDto> GetUserByUidAsync(string uid)
+    {
+        var user = await _context.Users.FirstOrDefaultAsync(u => u.Uid == uid);
+        
+        if (user == null)
+        {
+            throw new InvalidOperationException($"A user with uid {uid} is not found.");
+        }

```

Comment:
can be simplified with `??` construction
And also use `is null` comparison in the future